Bottom: 611090f1257cd83b8ccf4ba0711d09770b47f2b3
Top:    c8a8ae610747b3704dff06e3f9526bfb5f17dd59
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-12-04 20:31:13 +0000

Fix model to sit on top of copper tracks


---

diff --git a/src/hid/step/step.c b/src/hid/step/step.c
index 76ab121..f841e45 100644
--- a/src/hid/step/step.c
+++ b/src/hid/step/step.c
@@ -61,7 +61,7 @@
 
 
 #define HACK_BOARD_THICKNESS MM_TO_COORD(1.6)
-
+#define HACK_COPPER_THICKNESS MM_TO_COORD(0.035)
 
 HID step_hid;
 
@@ -408,9 +408,9 @@ step_do_export (HID_Attr_Val * options)
         instance->ox += COORD_TO_STEP_X (PCB, element->MarkX);
         instance->oy += COORD_TO_STEP_Y (PCB, element->MarkY);
 #ifdef REVERSED_PCB_CONTOURS
-        instance->oz += COORD_TO_STEP_Z (PCB, on_solder ? -HACK_BOARD_THICKNESS : 0);
+        instance->oz += COORD_TO_STEP_Z (PCB, on_solder ? -HACK_BOARD_THICKNESS - HACK_COPPER_THICKNESS : HACK_COPPER_THICKNESS);
 #else
-        instance->oz += COORD_TO_STEP_Z (PCB, on_solder_negate * -HACK_BOARD_THICKNESS / 2);
+        instance->oz += COORD_TO_STEP_Z (PCB, on_solder_negate * (-HACK_BOARD_THICKNESS / 2 -HACK_COPPER_THICKNESS));
 #endif
 
         model->instances = g_list_append (model->instances, instance);
